homeassistant/components/media_player/onkyo.py

```diff
@@ -82,6 +84,8 @@ class OnkyoDevice(MediaPlayerDevice):
 
     def __init__(self, receiver, sources, name=None):
         """Initialize the Onkyo Receiver."""
+        from concurrent.futures import ThreadPoolExecutor
```

Comment:
Oh no. This is waaay too heavy of a solution. If you would want 1 thread, you should start just a thread, not introduce a whole pool. However, even a thread is not allowed to be added. Imagine all our integrations include a thread, we would swamp the CPU. 

So instead, you should use asyncio and locks (Home Assistant core is based on asyncio). Asyncio tasks waiting for a lock don't take up any resources. It means that you need to convert all commands to their async variant and include the lock. Once lock has been acquired asynchronously, use the built-in worker pool to run a threaded command: `await hass.async_add_job(self.execute_command, command)`.

Reply:
Thanks for review. Good point, I'll try to rework.

If the other commit (SUPPORT_VOLUME_STEP) looks OK, I'll make it a separate PR.